src/Voron/PageLocator.cs

```diff
@@ -18,55 +17,28 @@ private struct PageData
             public Page Page;
             [FieldOffset(16)]
             public bool IsWritable;
+            [FieldOffset(24)]
+            public int Generation;
         }
 
         private const long Invalid = -1;
-        private LowLevelTransaction _tx;
+        private int _generation = 1;
 
-        private PageData* _cache;
-        private ByteString _cacheMemory;
+        private readonly PageData[] _cache = new PageData[CacheSize];
```

Comment:
Do you know the impact of moving it to managed memory? It looks fine at glance but I wonder if this can make noticeable difference on heavy loaded systems.

Reply:
Impact is minimal because we are accessing it through `Unsafe.Read` code and the size is backed in on construction